runtime/caml/misc.h

```diff
@@ -56,10 +56,6 @@
   #define CAMLdeprecated_typedef(name, type) typedef type name
 #endif
 
-#if defined(__GNUC__)                                           \
```

Comment:
Are you sure all compilers able to build OCaml 5 will support this? Are you sure e.g. ICC supports `_Pragma`?

Reply:
> Intel® C++ Compiler Classic (icc) is deprecated and was discontinued in the oneAPI 2024.0 release. Intel recommends that customers transition now to using the LLVM-based Intel® oneAPI DPC++/C++ Compiler (icx)

I've tried both `icc` and `icx` with this branch and they build it fine (expect ~10Go of download and 50Go of expanded disk space).

```sh
(cd "$(mktemp -d)" && docker build --ssh default --platform linux/amd64 -f - .) <<"EOF"
# syntax=docker/dockerfile:1
FROM intel/oneapi-basekit:2024-2.1-0-devel-ubuntu22.04
ADD --link git@github.com:MisterDA/ocaml.git#compiler-attributes /tmp/ocaml
WORKDIR /tmp/ocaml
RUN ./configure --disable-ocamldoc \
    CC=/opt/intel/oneapi/compiler/latest/bin/icx-cc
# checking C compiler vendor... clang-19-0
RUN make -j$(nproc)
EOF

(cd "$(mktemp -d)" && docker build --ssh default --platform linux/amd64 -f - .) <<"EOF"
# syntax=docker/dockerfile:1
FROM intel/oneapi-hpckit:2023.2.1-devel-ubuntu22.04
ADD --link git@github.com:MisterDA/ocaml.git#compiler-attributes /tmp/ocaml
WORKDIR /tmp/ocaml
RUN ./configure --disable-ocamldoc \
    CC=icc CFLAGS=-diag-disable=10441
# checking C compiler vendor... icc-2021
RUN make -j$(nproc)
EOF
```

…but your point remains valid, and I'll use `__GNUC__`, `__llvm__`, and `_MSC_VER` instead.

Reply:
`_Pragma` itself is C99, but its content are of course specific to each compiler.